src/GeositeFramework/plugins/layer_selector_v2/main.js

```diff
@@ -413,7 +413,13 @@ define([
                 // Fetch all map services found.
                 var promise = all(_.map(serviceUrls, function(v, serviceUrl) {
                     // Cache map service response.
-                    return ajaxUtil.fetch(serviceUrl);
+                    var options = {};
+                    if (serviceUrl.match(/WMS/i)) {
+                        options.format = 'text';
+                        options.content = '';
+                    }
+
+                    return ajaxUtil.fetch(serviceUrl, options);
```

Comment:
I think to account for minor differences between REST and WMS (and future services) we should start using those service classes to perform service operations. I'll try to think of a way to do this in the future (something like `createServiceFromUrl(serviceUrl).fetchMapService()` ?)